### PR TITLE
feat(content-answers): Upgrade Content Answers

### PR DESCRIPTION
**Pull request by greg-in-a-box**
Updated Content Answer with the latest features

### CODE REVIEW

File: src/elements/common/content-answers/ContentAnswersModal.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+// eslint-disable-next-line import/named
+import { ElementsXhrError } from '../../common/types/api';
+
+import messages from './messages';
+
+export interface ContentAnswersModalExternalProps {
+    isCitationsEnabled?: boolean;
+    isMarkdownEnabled?: boolean;
+    isResetChatEnabled?: boolean;
+    onAsk?: () => void;
```
[the remark this answers is not in the document]
Comment (greg-in-a-box) [reply src/elements/common/content-answers/ContentAnswersModal.tsx:35]: its separated like that because a customer can submit onAsk and onRequestClose from the root lvl via contentAnswersProps

File: src/elements/common/content-answers/ContentAnswersModal.tsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            onModalClose={handleOnRequestClose}
+            open={isOpen}
+            onOpenChange={handleOnRequestClose}
+            onClearAction={handleClearConversation}
+            questions={questions}
+            retryQuestion={handleRetry}
+            submitQuestion={handleAsk}
+            suggestedQuestions={suggestedQuestions || localizedQuestions}
+            warningNotice={spreadsheetNotice}
+            warningNoticeAriaLabel={formatMessage(messages.welcomeMessageSpreadsheetNoticeAriaLabel)}
```
[the remark this answers is not in the document]
Comment (greg-in-a-box) [reply src/elements/common/content-answers/ContentAnswersModal.tsx:189]: `warningNotice` controls if the ariaLabel is used.

File: src/elements/common/content-answers/ContentAnswersModal.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    currentUser,
+    file,
+    isOpen,
+    onAsk,
+    onClearConversation,
+    onRequestClose,
+    suggestedQuestions,
+    isCitationsEnabled = true,
+    isMarkdownEnabled = true,
+    isResetChatEnabled = true,
```
[the remark this answers is not in the document]
Comment (greg-in-a-box) [reply src/elements/common/content-answers/ContentAnswersModal.tsx:58]: we want them on to showcase the features and if the customers dont want them, they can turn them off.

File: src/elements/common/content-answers/ContentAnswersOpenButton.scss
[the remark this answers is not in the document]
Comment (greg-in-a-box) [reply src/elements/common/content-answers/ContentAnswersOpenButton.scss:1]: updated

File: src/elements/common/content-answers/ContentAnswersModal.tsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+    const handleSuccessCallback = useCallback((response: AxiosResponse): void => {
+        const question = {
+            answer: response.data.answer,
+            created_at: response.data.created_at,
+            citations: response.data.citations,
+            error: null,
+            isCompleted: true,
+            isLoading: false,
+        };
```
[the remark this answers is not in the document]
Comment (greg-in-a-box) [reply src/elements/common/content-answers/ContentAnswersModal.tsx:81]: updated with a const updatedLastQuestion to be more specific

File: src/elements/common/content-answers/__tests__/ContentAnswers.test.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        renderComponent();
+
+        const button = screen.getByRole('button', { name: 'Box AI' });
+        await userEvent.click(button);
+
+        const modal = screen.getByTestId('content-answers-modal');
+        expect(modal).toBeInTheDocument();
+
+        const textArea = screen.getByRole('textbox', { name: 'Ask anything about this doc' });
+        fireEvent.change(textArea, { target: { value: prompt } });
```
Comment (tjiang-box) [inline src/elements/common/content-answers/__tests__/ContentAnswers.test.tsx:42]: the `prompt` here seems not defined in this file. 

And could we try using `userEvent.type(textArea, 'question prompt')`. Same for ones used in ContentAnswersModal.test.tsx file. 
Example https://github.com/box/box-ui-elements/blob/master/src/elements/content-explorer/__tests__/RenameDialog.test.tsx#L59-L62

File: src/elements/common/content-answers/ContentAnswers.tsx [hunk 6]
```diff
@@ -1,21 +1,26 @@
...
 import { BoxItem } from '../../common/types/core';
 
-type ExternalProps = {
+type ExternalProps = ContentAnswersModalExternalProps & {
     show?: boolean;
 };
 
 type Props = {
     file: BoxItem;
 };
```
Comment (tjiang-box) [inline src/elements/common/content-answers/ContentAnswers.tsx:16]: should the prop types above changed to interface like below
```
export interface contentAnswersProps extends ContentAnswersModalExternalProps {
    show?: boolean;
    file: BoxItem;
}
```

**Review by tjiang-box — CHANGES_REQUESTED** [review]

File: src/elements/common/content-answers/ContentAnswersModal.tsx [hunk 7]
```diff
@@ -177,10 +183,10 @@ const ContentAnswersModal = ({
             isCitationsEnabled={isCitationsEnabled}
             isMarkdownEnabled={isMarkdownEnabled}
             isResetChatEnabled={isResetChatEnabled}
+            onClearAction={handleClearConversation}
+            onOpenChange={handleOnRequestClose}
             onModalClose={handleOnRequestClose}
```
Comment (benjamin-shen) [inline src/elements/common/content-answers/ContentAnswersModal.tsx:188]: m comes before o in the alphabet!

File: src/elements/common/content-answers/ContentAnswers.tsx
Comment (tjuanitas) [inline src/elements/common/content-answers/ContentAnswers.tsx:1]: it looks like this file should be export two separate types?

one interface for external props - configuration options through ContentPreview
one interface for TS reason - regular component props

File: src/elements/common/content-answers/ContentAnswersOpenButton.scss [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    background: tokens.$border-cta-border-ai;
+    border: 1px solid tokens.$white-opacity-30;
+}
+
+@mixin BackgroundGradientHover {
+    background: linear-gradient(0deg, #2223, #2223), linear-gradient(135deg, #fe01da 0%, tokens.$light-blue-100 100%);
+    border: 1px solid #2222224d;
+}
+
+.bdl-ContentAnswersOpenButton {
```
Comment (tjuanitas) [inline src/elements/common/content-answers/ContentAnswersOpenButton.scss:13]: for components in the elements folder,  i think we want to use the `be` prefix instead of `bdl`

be = box-elements
bdl = box-design-language

i.e. `be-ContentAnswersOpenButton`

File: src/elements/common/content-answers/ContentAnswersOpenButton.scss [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+            box-shadow:
+                0 0 0 1px tokens.$gray-white,
+                0 0 0 3px tokens.$light-blue-100;
+        }
+    }
+
+    .bdl-ContentAnswersOpenButton-tooltip {
+        text-align: center;
+    }
```
Comment (tjuanitas) [inline src/elements/common/content-answers/ContentAnswersOpenButton.scss:47]: is this class used?

File: src/elements/content-preview/ContentPreview.js [hunk 10]
```diff
@@ -1319,75 +1320,88 @@ class ContentPreview extends React.PureComponent<Props, State> {
...
-                        )}
-                        <div className="bcpr-body">
-                            <div className="bcpr-container" onMouseMove={this.onMouseMove} ref={this.containerRef}>
-                                {file && (
-                                    <Measure bounds onResize={this.onResize}>
-                                        {({ measureRef: previewRef }) => (
-                                            <div ref={previewRef} className="bcpr-content" />
+                    <Notification.Provider>
+                        <Notification.Viewport />
+                        <TooltipProvider>
```
Comment (tjuanitas) [inline src/elements/content-preview/ContentPreview.js:1325]: internal apps will likely have these providers already. any concerns with nested providers?

File: src/elements/common/content-answers/ContentAnswersOpenButton.scss [hunk 11]
```diff
@@ -0,0 +1,48 @@
+@use '@box/blueprint-web-assets/tokens/tokens.scss';
+
+@mixin BackgroundGradient {
+    background: tokens.$border-cta-border-ai;
+    border: 1px solid tokens.$white-opacity-30;
+}
+
+@mixin BackgroundGradientHover {
+    background: linear-gradient(0deg, #2223, #2223), linear-gradient(135deg, #fe01da 0%, tokens.$light-blue-100 100%);
+    border: 1px solid #2222224d;
```
[the remark this answers is not in the document]
Comment (greg-in-a-box) [reply src/elements/common/content-answers/ContentAnswersOpenButton.scss:10]: no idea, color wise they are the same that is being used in other places

File: src/elements/common/content-answers/ContentAnswersModal.tsx [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+            submitQuestion={handleAsk}
+            suggestedQuestions={suggestedQuestions || localizedQuestions}
+            warningNotice={spreadsheetNotice}
+            warningNoticeAriaLabel={formatMessage(messages.welcomeMessageSpreadsheetNoticeAriaLabel)}
+            userInfo={userInfo}
+        />
+    );
+};
+
+export default withAPIContext(withCurrentUser(ContentAnswersModal));
```
[the remark this answers is not in the document]
Comment (greg-in-a-box) [reply src/elements/common/content-answers/ContentAnswersModal.tsx]: yea we need this here for the name and avatar url

File: src/elements/common/content-answers/ContentAnswers.tsx [hunk 13]
```diff
@@ -29,32 +34,38 @@ const ContentAnswers = ({ file }: Props) => {
...
                 onAsk={handleAsk}
                 onRequestClose={handleClose}
+                {...rest}
             />
         </div>
     );
 };
 
-export type ContentAnswersProps = ExternalProps;
+export type ContentAnswers = ExternalProps & Props;
```
Comment (tjuanitas) [inline src/elements/common/content-answers/ContentAnswers.tsx:70]:
```suggestion
export type ContentAnswersProps = ExternalProps & Props;
```

File: src/elements/common/content-answers/ContentAnswers.tsx [hunk 14]
```diff
@@ -1,21 +1,26 @@
...
-};
+}
 
-const ContentAnswers = ({ file }: Props) => {
+const ContentAnswers = ({
+    file,
+    onAsk,
+    onRequestClose,
+    ...rest
+}: Omit<ContentAnswersModalExternalProps & Props, 'show'>) => {
```
Comment (tjuanitas) [inline src/elements/common/content-answers/ContentAnswers.tsx:23]: `show` is not included in `ContentAnswersModalExternalProps`

File: src/elements/content-preview/README.md [hunk 15]
```diff
@@ -33,23 +33,24 @@
...
+| responseInterceptor  | function | | *See the [developer docs](https://developer.box.com/docs/box-content-preview#section-options).*                                                                                     |
+| sharedLink           | string |  | *See the [developer docs](https://developer.box.com/docs/box-content-preview#section-options).*                                                                                     |
+| sharedLinkPassword   | string |  | *See the [developer docs](https://developer.box.com/docs/box-content-preview#section-options).*                                                                                     |
+| showAnnotations      | boolean | `true` | *See the [developer docs](https://developer.box.com/docs/box-content-preview#section-options).*                                                                                     |
+| shouldProvide        | boolean | `true` | Decide if it should wrap the element with the `Notification` and `Tooltip` Provider. Set this to false if you already wrap your app with both of these providers.                       |
```
Comment (tjuanitas) [inline src/elements/content-preview/README.md:54]: if this is a boolean then maybe `hasProviders`?

---

alternate approach:
or what about an array/object that specifies providers? that way it's more configurable depending on the providers that the consumer already has

for example:

```
// in consuming app
const providers = ['date-picker', 'notification', 'tooltip']; // Blueprint DatePicker has a provider we'll eventually need to include

<ContentPreview providers={providers} ... />


// in new providers component
const ElementProviders = ({ children, providers = ['notification', 'tooltip'] }) => {
    let element = children;

    providers.forEach(provider => {
        switch (provider) {
            case 'date-picker':
                ...
                break;
            case 'notification':
                element = <Notification.Provider>...</Notification.Provider>;
                break;
            case 'tooltip':
                element = <TooltipProvider>{element}</TooltipProvider>;
                break;
        }
    });

    return element;
};
```

File: src/elements/content-preview/preview-header/PreviewHeader.js [hunk 16]
```diff
@@ -62,7 +62,8 @@ const PreviewHeader = ({
     token,
 }: Props) => {
     const fileId = file && file.id;
-    const shouldRenderAnswers = fileId && contentAnswersProps.show;
+    const { show: showContentAnswers, ...restContentAnswersProps } = contentAnswersProps;
+    const shouldRenderAnswers = fileId && showContentAnswers;
```
Comment (tjuanitas) [inline src/elements/content-preview/preview-header/PreviewHeader.js:66]: i don't see harm in passing `show` along with the rest of the contentAnswersProps to ContentAnswers. it could simplify the exporting types and whatnot

File: src/api/Intelligence.js [hunk 17]
```diff
@@ -16,11 +17,19 @@ class Intelligence extends Base {
     /**
      * API endpoint to ask ai a question
      *
-     * @param {string} prompt - Question
+     * @param question
      * @param {Array<object>} items - Array of items to ask about
+     * @param dialogueHistory
+     * @param options
```
Comment (tjuanitas) [inline src/api/Intelligence.js:23]: can we include types of these new params?

File: src/elements/common/content-answers/ContentAnswersModal.tsx [hunk 18]
```diff
@@ -0,0 +1,202 @@
...
+    isCitationsEnabled = true,
+    isMarkdownEnabled = true,
+    isResetChatEnabled = true,
+}: ContentAnswersModalProps) => {
+    const { formatMessage } = useIntl();
+    const [isLoading, setIsLoading] = useState<boolean>(false);
+    const [questions, setQuestions] = useState<QuestionType[]>([]);
+    let localizedQuestions: SuggestedQuestionType[] = [];
+
+    if (!suggestedQuestions) {
```
Comment (tjuanitas) [inline src/elements/common/content-answers/ContentAnswersModal.tsx:66]: fyi an empty array for `suggestedQuestions` is truthy so this only enters the `if` statement for `undefined`. is that okay?

my guess is that it could render empty space but that seems out of scope for this component and more of something that should be handled by the shared-feature

File: src/elements/common/content-answers/ContentAnswersModal.tsx [hunk 19]
```diff
@@ -0,0 +1,202 @@
...
+        localizedQuestions = DOCUMENT_SUGGESTED_QUESTIONS.map(question => ({
```
Comment (tjuanitas) [inline src/elements/common/content-answers/ContentAnswersModal.tsx:67]: is this something the webapp is doing for all file extensions?

File: src/elements/common/content-answers/ContentAnswersModal.tsx
Comment (tjuanitas) [inline src/elements/common/content-answers/ContentAnswersModal.tsx:1]: i feel like uses of useCallback in this file are unnecessary:
https://react.dev/reference/react/useCallback#should-you-add-usecallback-everywhere

File: src/elements/common/content-answers/ContentAnswersModal.tsx [hunk 20]
```diff
@@ -0,0 +1,202 @@
...
+
+    const handleOnRequestClose = useCallback(() => {
+        if (onRequestClose) {
+            onRequestClose();
+        }
+    }, [onRequestClose]);
+
+    const fileName = getProp(file, 'name');
+    const currentFileExtension = getProp(file, 'extension');
+    const userInfo = { name: getProp(currentUser, 'name') || '', avatarURL: getProp(currentUser, 'avatarURL') || '' };
```
Comment (tjuanitas) [inline src/elements/common/content-answers/ContentAnswersModal.tsx:174]: `getProp` takes a third parameter for default values instead of needing `||`

File: src/elements/common/content-answers/ContentAnswersModal.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (tjuanitas) [reply src/elements/common/content-answers/ContentAnswersModal.tsx:202]: i think we talked about this but can you remind me what's happening here? shouldn't the props forwarded from withAPIContext be passed through withCurrentUser to this component?

File: src/elements/common/content-answers/ContentAnswersOpenButton.scss [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+@mixin BackgroundGradientHover {
+    background: linear-gradient(0deg, #2223, #2223), linear-gradient(135deg, #fe01da 0%, tokens.$light-blue-100 100%);
+    border: tokens.$border-1 solid #2222224d;
+}
+
+.be-ContentAnswersOpenButton {
+    width: tokens.$size-20;
+    height: tokens.$size-20;
```
Comment (tjuanitas) [inline src/elements/common/content-answers/ContentAnswersOpenButton.scss:15]: is this correct/used? isn't size-20 like 80 pixels?

File: src/elements/common/content-answers/ContentAnswersModal.tsx (hunk 19 above)
Comment (greg-in-a-box) [reply src/elements/common/content-answers/ContentAnswersModal.tsx:67]: yes

File: .storybook/modes.ts [hunk 22]
```diff
@@ -0,0 +1,13 @@
+export const allModes = {
+    large: {
+        viewport: 'l',
+    },
+    // Note, you can still specify the more
+    // specific options listed in the section above
```
Comment (tjuanitas) [inline .storybook/modes.ts:6]: what does "section above" refer to?

Comment (tjuanitas) [reply .storybook/modes.ts:6]: it looks like this file/object is used once, what about adding `viewport` property directly to preview.tsx?

File: src/elements/common/Provide.tsx
Comment (tjuanitas) [inline src/elements/common/Provide.tsx:1]: nit: `Provide` sounds a little odd to me as a component because of the verb. `Providers` sounds a little clearer

Comment (tjuanitas) [reply src/elements/common/Provide.tsx:1]: related to conversation about granular providers, we might need to consider how customers can enable/disable providers like Theme, Router (react-router upgrade)

File: src/elements/common/__tests__/Provider.test.tsx [hunk 23]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { render } from '../../../test-utils/testing-library';
+import Provide from '../Provide';
+
+describe('Provide component', () => {
```
Comment (tjuanitas) [inline src/elements/common/__tests__/Provider.test.tsx:5]: should be path to component

File: src/elements/content-preview/README.md [hunk 24]
```diff
@@ -33,23 +33,24 @@
...
+| responseInterceptor  | function | | *See the [developer docs](https://developer.box.com/docs/box-content-preview#section-options).*                                                                                     |
+| sharedLink           | string |  | *See the [developer docs](https://developer.box.com/docs/box-content-preview#section-options).*                                                                                     |
+| sharedLinkPassword   | string |  | *See the [developer docs](https://developer.box.com/docs/box-content-preview#section-options).*                                                                                     |
+| showAnnotations      | boolean | `true` | *See the [developer docs](https://developer.box.com/docs/box-content-preview#section-options).*                                                                                     |
+| hasProviders         | boolean | `true` | Decide if it should wrap the element with the `Notification` and `Tooltip` Provider. Set this to false if you already wrap your app with both of these providers.                       |
```
Comment (tjuanitas) [inline src/elements/content-preview/README.md:54]: nit: 🔤

File: src/elements/content-preview/ContentPreview.js [hunk 25]
```diff
@@ -121,6 +122,7 @@ type Props = {
     responseInterceptor?: Function,
     sharedLink?: string,
     sharedLinkPassword?: string,
+    hasProviders?: boolean,
```
Comment (tjuanitas) [inline src/elements/content-preview/ContentPreview.js:125]: nit: 🔤

File: src/elements/common/content-answers/ContentAnswersModal.tsx [hunk 26]
```diff
@@ -0,0 +1,199 @@
...
+            isLoading: false,
+        };
+
+        setQuestions(prevState => {
+            return [...prevState.slice(0, -1), errorQuestion];
+        });
+    }, []);
+
+    const handleAsk = useCallback(
+        async (question: QuestionType, aiAgent: AgentType, isRetry = false) => {
```
Comment (tjuanitas) [inline src/elements/common/content-answers/ContentAnswersModal.tsx:107]: what's `aiAgent` for?

File: src/elements/common/content-answers/ContentAnswersModal.tsx [hunk 27]
```diff
@@ -0,0 +1,199 @@
...
+    }, [onClearConversation]);
+
+    const handleOnRequestClose = useCallback(() => {
+        if (onRequestClose) {
+            onRequestClose();
+        }
+    }, [onRequestClose]);
+
+    const fileName = getProp(file, 'name');
+    const currentFileExtension = getProp(file, 'extension');
```
Comment (tjuanitas) [inline src/elements/common/content-answers/ContentAnswersModal.tsx:170]: nit: `current` seems unneeded
```suggestion
    const fileExtension = getProp(file, 'extension');
```

File: src/elements/common/content-answers/__tests__/ContentAnswersModal.test.tsx [hunk 28]
```diff
@@ -0,0 +1,172 @@
...
+        expect(headerIcon).toBeInTheDocument();
+    });
+
+    test('should ask for answer when prompt is submitted', async () => {
+        const onAskMock = jest.fn();
+        const { answer = '', prompt } = mockQuestionsWithAnswer[0];
+        renderComponent(mockApi, { onAsk: onAskMock });
+
+        const textArea = screen.getByRole('textbox', { name: 'Ask anything about this doc' });
+        fireEvent.change(textArea, { target: { value: prompt } });
```
Comment (tjuanitas) [inline src/elements/common/content-answers/__tests__/ContentAnswersModal.test.tsx:52]: nit: i think uses of `fireEvent.change` could be `userEvent.type`?

File: src/elements/common/__tests__/Provider.test.tsx [hunk 29]
```diff
@@ -0,0 +1,47 @@
...
+            <Provide hasProviders={true}>
+                <div>Child Content</div>
+            </Provide>,
+        );
+
+        expect(screen.getByText('Child Content')).toBeInTheDocument();
+        expect(screen.getByRole('region', { name: 'Notifications (F8)' })).toBeInTheDocument();
+    });
+
+    test('renders only children when shouldProvide is false', () => {
```
Comment (tjuanitas) [inline src/elements/common/__tests__/Provider.test.tsx:17]:
```suggestion
    test('renders only children when hasProviders is false', () => {
```

File: src/elements/common/__tests__/Provider.test.tsx [hunk 30]
```diff
@@ -0,0 +1,47 @@
...
+        const screen = render(
+            <Provide hasProviders={false}>
+                <div>Child Content</div>
+            </Provide>,
+        );
+        expect(screen.getByText('Child Content')).toBeInTheDocument();
+        expect(screen.queryByRole('region', { name: 'Notifications (F8)' })).not.toBeInTheDocument();
+    });
+
+    test('throws an error if more than one child is provided when shouldProvide is false', () => {
```
Comment (tjuanitas) [inline src/elements/common/__tests__/Provider.test.tsx:27]:
```suggestion
    test('throws an error if more than one child is provided when hasProviders is false', () => {
```

File: src/elements/common/__tests__/Provider.test.tsx [hunk 31]
```diff
@@ -0,0 +1,47 @@
...
+            render(
+                <Provide hasProviders={false}>
+                    <div>Child 1</div>
+                    <div>Child 2</div>
+                </Provide>,
+            ),
+        ).toThrow();
+    });
+
+    test('renders children within Notification and TooltipProvider by default when shouldProvide is not provided', () => {
```
Comment (tjuanitas) [inline src/elements/common/__tests__/Provider.test.tsx:38]:
```suggestion
    test('renders children within Notification and TooltipProvider by default when hasProviders is not provided', () => {
```

File: src/elements/common/content-answers/__tests__/ContentAnswersOpenButton.test.tsx [hunk 32]
```diff
@@ -0,0 +1,81 @@
...
+        });
+        expect(tooltip).toBeInTheDocument();
+    });
+
+    test('should not call onclick callback when filetype is not allowed', async () => {
+        const onClick = jest.fn();
+        renderComponent({ fileExtension: 'invalid', onClick });
+        expect(onClick).toBeCalledTimes(0);
+        const button = screen.getByRole('button');
+        await userEvent.click(button, { pointerEventsCheck: 0 });
```
Comment (tjuanitas) [inline src/elements/common/content-answers/__tests__/ContentAnswersOpenButton.test.tsx:64]: what's `pointerEventsCheck` for? is this optimization?

File: src/elements/common/content-answers/__tests__/ContentAnswersOpenButton.test.tsx [hunk 33]
```diff
@@ -0,0 +1,81 @@
...
+        const button = screen.getByRole('button', { name: 'Box AI' });
+        await userEvent.hover(button);
+        const tooltip = await screen.findByRole('tooltip', { name: messages.defaultTooltip.defaultMessage });
+        expect(tooltip).toBeInTheDocument();
+    });
+
+    test('should display not allowed tooltip', async () => {
+        renderComponent({ fileExtension: 'invalid' });
+        const button = screen.getByRole('button', { name: 'Box AI' });
+        await userEvent.hover(button, { pointerEventsCheck: 0 });
```
Comment (tjuanitas) [inline src/elements/common/content-answers/__tests__/ContentAnswersOpenButton.test.tsx:42]: is `hover` using pointer events?

File: src/elements/common/content-answers/__tests__/ContentAnswersOpenButton.test.tsx (hunk 32 above)
Comment (greg-in-a-box) [reply src/elements/common/content-answers/__tests__/ContentAnswersOpenButton.test.tsx:64]: it throws an error like this `Error: Unable to perform pointer interaction as the element has `pointer-events: none`:` if that flag isnt there, the button from blueprint set `pointer-events` to `none`

File: src/elements/common/content-answers/__tests__/ContentAnswersOpenButton.test.tsx (hunk 33 above)
Comment (greg-in-a-box) [reply src/elements/common/content-answers/__tests__/ContentAnswersOpenButton.test.tsx:42]: same as above

File: src/elements/common/content-answers/ContentAnswersModal.tsx (hunk 26 above)
Comment (greg-in-a-box) [reply src/elements/common/content-answers/ContentAnswersModal.tsx:107]: optional option that is not present enabled publicly

File: src/elements/common/Provide.tsx
Comment (greg-in-a-box) [reply src/elements/common/Provide.tsx:1]: yes for theme, but router no since that is more universal, unless the backwards compat guide says we need one

**Review by tjuanitas — APPROVED** [review]
lgtm

File: src/api/Intelligence.js [hunk 34]
```diff
@@ -17,10 +17,10 @@ class Intelligence extends Base {
     /**
      * API endpoint to ask ai a question
      *
-     * @param question
+     * @param {QuestionType} question - Object should at least contain the prompt, which is the question to ask
      * @param {Array<object>} items - Array of items to ask about
-     * @param dialogueHistory
-     * @param options
+     * @param {Array<QuestionType>} dialogueHistory - Array of previous questions object that already have answers
```
Comment (benjamin-shen) [inline src/api/Intelligence.js:22]: naming `QuestionType` with a `-Type` suffix instead of `Question` is like naming something `FooInterface` instead of `Foo`, right? is that an existing naming convention in this repo? if not, can we rename `QuestionType` to `Question`?

is it possible to make these types stronger, eg. define a `QuestionWithAnswer` type that is `Question & { answer... }`

File: src/elements/common/Providers.js.flow [hunk 35]
```diff
@@ -0,0 +1,23 @@
...
+import * as React from 'react';
+import { Notification, TooltipProvider } from '@box/blueprint-web';
+
+export interface ProvideProps {
+    children: React.Node;
+    hasProviders: ?boolean;
+}
+
+const Provide = ({ children, hasProviders }: ProvideProps) => {
+    if (!!hasProviders) {
```
Comment (benjamin-shen) [inline src/elements/common/Providers.js.flow:11]: why is the `!!` necessary?

File: src/elements/content-preview/stories/tests/ContentPreview-visual.stories.js [hunk 36]
```diff
@@ -143,7 +143,7 @@ export const citationDisabled = {
         const answer = await modal.findByText('Public APIs are important because of key and important reasons.');
         expect(answer).toBeInTheDocument();
 
-        expect(modal.getByText('Based on:')).toBeInTheDocument();
+        expect(modal.queryByText('Based on:')).not.toBeInTheDocument();
```
Comment (benjamin-shen) [inline src/elements/content-preview/stories/tests/ContentPreview-visual.stories.js:146]: where is the text `'Based on:'` actually rendered from? does it come from the api response or is it hardcoded in the repo?

File: src/api/Intelligence.js (hunk 34 above)
Comment (greg-in-a-box) [reply src/api/Intelligence.js:22]: that type is coming from the shared-feature, its the name they are using

File: src/elements/content-preview/stories/tests/ContentPreview-visual.stories.js (hunk 36 above)
Comment (greg-in-a-box) [reply src/elements/content-preview/stories/tests/ContentPreview-visual.stories.js:146]: part of the citation feature that appears in the answer if the citations are present via the shared-features

**Review by benjamin-shen — APPROVED** [review]

**Review by tjuanitas — APPROVED** [review]
